src/app/modules/workspace/__tests__/workspace.routes.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { ObjectId } from 'bson-ext'
+import mongoose from 'mongoose'
+import supertest, { Session } from 'supertest-session'
+
+import { getWorkspaceModel } from 'src/app/models/workspace.server.model'
+import { WorkspacesRouter } from 'src/app/routes/api/v3/admin/workspaces'
+
+import {
+  createAuthedSession,
+  logoutSession,
+} from 'tests/integration/helpers/express-auth'
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+import { jsonParseStringify } from 'tests/unit/backend/helpers/serialize-data'
+
+const WorkspaceModel = getWorkspaceModel(mongoose)
+
+const app = setupApp('/workspaces', WorkspacesRouter, {
+  setupWithAuth: true,
+})
+
+const MOCK_USER_ID = new ObjectId()
+const MOCK_FORM_ID = new ObjectId()
+const MOCK_WORKSPACE_ID = new ObjectId()
+const MOCK_WORKSPACE_FIELDS = {
+  _id: MOCK_WORKSPACE_ID,
+  title: 'Workspace1',
+  admin: MOCK_USER_ID,
+  formIds: [],
+}
+
+describe('workspaces.routes', () => {
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => {
+    request = supertest(app)
+    const { user } = await dbHandler.insertEncryptForm({
+      formId: MOCK_FORM_ID,
+      userId: MOCK_USER_ID,
+    })
+    request = await createAuthedSession(user.email, request)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('GET /workspaces', () => {
+    const GET_WORKSPACES_ENDPOINT = '/workspaces'
+
+    it('should return 200 with an empty array when a user has no workspaces', async () => {
+      const response = await request.get('/workspaces')
+
+      expect(response.status).toEqual(200)
+      expect(response.body).toEqual([])
+    })
+
+    it("should return 200 with an array of the user's workspaces", async () => {
+      await WorkspaceModel.create(MOCK_WORKSPACE_FIELDS)
+      const response = await request.get(GET_WORKSPACES_ENDPOINT)
+      const expected = await WorkspaceModel.find({ _id: MOCK_WORKSPACE_ID })
```

Comment:
nit: would it be clearer for the reader if we used `MOCK_WORKSPACE_DOC._id` (even though they are the same thing, haha)

Reply:
Yep, I think so, will update to use `MOCK_WORKSPACE_DOC._id` instead